xarray/core/npcompat.py

```diff
@@ -58,3 +59,43 @@ def isdtype(dtype, kind):
             return any(isinstance(dtype, kind) for kind in translated_kinds)
         else:
             return any(np.issubdtype(dtype, kind) for kind in translated_kinds)
+
+
```

Comment:
`npcompat` currently (and historically) only has direct back-ports of NumPy features (hence the numpy license note).

So I would suggest keeping this new stuff elsewhere -- maybe in a new file or in the previous location?

Reply:
How about `array_api_compat.py`? The main idea is to make it very clear that we eventually want to remove `_future_array_api_result_type` and the `result_type` wrapper. Additionally, if we kept it in `dtypes`, we'd have to find a name other than `result_type` (since that's already taken).

Reply:
That sounds perfect!